src-ghc/Pact/PersistPactDb/Regression.hs

```diff
@@ -33,11 +33,12 @@ loadModule = do
   let mn = ModuleName "simple" Nothing
   case r of
     Left a -> throwFail $ "module load failed: " ++ show a
-    Right _ -> case preview (rEvalState . evalRefs . rsNewModules . ix mn) s of
-      Just md -> case traverse (traverse toPersistDirect) md of
+    Right _ -> case preview (rEvalState . evalRefs . rsLoadedModules . ix mn) s of
+      Just (md,_) -> case traverse (traverse toPersistDirect) md of
         Right md' -> return (mn,md,md')
         Left e -> throwFail $ "toPersistDirect failed: " ++ show e
-      Nothing -> throwFail $ "Failed to find module 'simple': " ++ show (view (rEvalState . evalRefs . rsNewModules) s)
+      Nothing -> throwFail $ "Failed to find module 'simple': " ++
+        show (view (rEvalState . evalRefs . rsLoadedModules) s)
```

Comment:
We're using this pattern `traverse (traverse toPersistDirect)` alot. This treats `ModuleData` like a composition of functors - it may be worth newtyping over the composite and deriving some instances for it so we can reduce the duplication.

Reply:
> it may be worth newtyping over the composite and deriving some instances for it so we can reduce the duplication
I don't know what this means, can you give an example?

Reply:
i'm thinking this looks very similar to the traversable instance for `Compose`, but i'm stuck on which path the `Traversable` instance derived for `ModuleData` actually traverses down. Is it down `_mdRefMap`? Both? I'll figure this out later. It's not necessary.